lib/vdsm/virt/vmstats.py

```diff
@@ -130,6 +130,7 @@ def cpu(stats, first_sample, last_sample, interval):
                 ((last_sample['cpu.time'] - first_sample['cpu.time']) -
                  cpu_sys),
                 interval)
+            stats['cpuUser'] = 0.00 if stats['cpuUser'] < 0 else stats['cpuUser']
```

Comment:
Thank you for the fix.  I'd suggest to polish it a bit:

- Wouldn't it be better to handle it in `_usage_percentage`?  It would also handle a similar case for `stats['cpuSys']` if it can ever happen.
- It would be good to add an explanatory source code comment.
- I think it's better to check whether the value is close to zero or clearly negative and log a warning in the latter case, to not hide cases where the value would be clearly wrong.
- Using `0.0` rather than `0.00` would be consistent with the surrounding code.

Reply:
Could you please respond to the points above? You're free to disagree with them but we need to come to some conclusion in order to proceed.
As for the source code comment, it should explain, rather than the obvious thing of avoiding the negative value, why the value can become negative (something in the sense of the fix description you provided in this PR).
Also, when changing commits inside a Vdsm PR, we use force push rather than adding additional commits, so please squash the commits.

Reply:
Sorry, I had pasted the answers earlier too but they were in pending state.

Reply:
Also made the suggested changes in the comment and squashed the two commits.

Reply:
Thank you for clarification and improving the comment, the example was helpful.
One last thing to clarify is the "0.00" value: Since it's a float and not a string there, I fail to understand how the notation can influence the output. For example, in the Python interpreter prompt:

```
>>> x = 0.00
>>> x
0.0
```

What do I miss?

Reply:
Yes, thanks. I checked again, variable type is float. Made the change to "0.0".